aiohttp/signals.py

```diff
@@ -0,0 +1,29 @@
+import asyncio
+
+
+class Signal(list):
+    """
+    Coroutine-based signal implementation
+
+    To connect a callback to a signal, use any list method. If wish to pass
+    additional arguments to your callback, use :meth:`functools.partial`.
+
+    Signals are fired using the :meth:`send` coroutine, which takes named
+    arguments.
+    """
+
+    @asyncio.coroutine
+    def send(self, **kwargs):
+        """
+        Sends data to all registered receivers.
+        """
+        for receiver in self:
+            res = receiver(**kwargs)
```

Comment:
Initial version from @alexsdutton had signature check. But I quite unhappy with it.
I prefer adding good checker later but keeping no checks at all in first implementation.